### PR TITLE
Add endpoint for getting unassigned orders

### LINKED ISSUE #10: Endpoint to be called by Riders to list unassigned orders
Thinking a little bit more about how the demo apps can best assist a QA engineer when testing, as well as show something a bit more realistic when used for customer demos, this backend service could offer another interface.

It would:

1. handle `GET` requests to `/orders`
1. only return anything but `401 Unauthorized` to `rider` users
1. query the Firestore `orders` collection for documents where the `riderUsername` field is not populated (presumably missing, given the way the data structures are created at the time of writing this, but ultimately probably needs to also cover the case that this field has explicitly been set to `null` as that could become a valid document state in future)
1. return an array of order ids, nested within a map (key name `orderIds` would make sense) to align with [our promise](https://github.com/ably/asset-tracking-backend-demo#rest-api) that all endpoints return an `object` at root

Allow the demo app to have a workflow where the Rider sees a list of orders that need to be delivered, from which they can select one, for then a subsequent call to [Assign Order](https://github.com/ably/asset-tracking-backend-demo#assign-order) to take place.

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-2443) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by Peter-Maguire**
Resolves #10

### CODE REVIEW

File: functions/Handlers.js [hunk 1]
```diff
@@ -88,6 +88,8 @@ exports.createOrder = async (req, res, next) => {
       customerUsername: username,
       from,
       to,
+      // riderUsername has to be explicitly set to null as firestore cannot select by missing fields
+      riderUsername: null,
```
Comment (QuintinWillison) [inline functions/Handlers.js:92]: Looking at this with fresh eyes (further to my original issue analysis) perhaps cleaner to just delete the field?
https://stackoverflow.com/a/46984847/392847

**Review by QuintinWillison — CHANGES_REQUESTED** [review]
Within the scope of this pull request, we also need to add this new endpoint to the [API reference](https://github.com/ably/asset-tracking-backend-demo#rest-api).

File: functions/Handlers.js (hunk 1 above)
Comment (Peter-Maguire) [reply functions/Handlers.js:92]: If `riderUsername` is deleted, then it becomes impossible to query by orders which have no rider, as firestore has no way to query by fields that are missing

Comment (QuintinWillison) [reply functions/Handlers.js:92]: Ah, I see. Thanks for explaining. 😁

**Review by QuintinWillison — APPROVED** [review]

**Pull request merged**